src/dune_rules/expander.ml

```diff
@@ -663,6 +663,13 @@ let expand_pform_macro
            let* artifacts_host = t.artifacts_host in
            let+ b = Artifacts.binary_available artifacts_host s in
            b |> string_of_bool |> string))
+  | File_available ->
+    Direct (
+      Without
+        (let open Memo.O in
+         let file_path = Path.of_string s in
+         let+ available = Fs.file_exists file_path in
```

Comment:
This isn't going to work if `file_path` is a target of a rule. Is that intentional?

Reply:
I think I might have this working I implemented added a simple test to the same `file-available.t` test file.